metaflow/system/metaflow_system_monitor.py

```diff
@@ -0,0 +1,121 @@
+import sys
+from contextlib import contextmanager
+from typing import Optional, Union
+
+
+class SystemMonitor(object):
+    def __init__(self):
+        self._monitor = None
+        self._flow_name = None
+
+    def __del__(self):
+        if self._flow_name == "not_a_real_flow":
+            self.monitor.terminate()
+
+    def init_environment_outside_flow(
```

Comment:
this method does not depend on self. I would move it to a util or something since it is common across both logger and monitor.